engine/execution/ingestion/engine.go

```diff
@@ -558,7 +562,7 @@ func (e *Engine) executeBlock(ctx context.Context, executableBlock *entity.Execu
 
 	isExecutedBlockSealed := executableBlock.Block.Header.Height <= lastSealed.Height
 	broadcasted := false
-	if !isExecutedBlockSealed {
+	if !isExecutedBlockSealed && e.staker.AmIStakedAt(executableBlock.Block.ID()) {
```

Comment:
the block ID has been computed a few times, can we cache it?

Reply:
changed